### PR TITLE
Always release to vcpkg daily branch

### PR DESCRIPTION
**Pull request by danieljurek**
Fixes https://github.com/Azure/azure-sdk-for-cpp/issues/4422

Always pushes vcpkg up to the daily branch regardless of build/test results. The majority of skips of this job were from failures of flaky live tests, image problems, network/service problems, or other issues not relating to code defects. 

image

### CODE REVIEW

**Review by antkmsft — APPROVED** [review]

**Comment by antkmsft** [comment]
Not sure if this is a result of you running the changes in this PR, but FWIW, vcpkg daily validation PR (https://github.com/microsoft/vcpkg/pull/17119) looks very good now - it includes changes for all the packages, has changes for 43 files, and we can see that all of them do build successfully in the vcpkg CI (* - at this very moment, linux pipeline there is still pending, but all others are successful, including ARM and UWP, which typically have the most problems). I.e. the PR serves the goal: I can go there and see that we are in good shape to release to vcpkg.
image

**Comment by antkmsft** [comment]
And at this moment, Storage packages are missing from the PR, I don't understand.

File: eng/pipelines/templates/stages/platform-matrix-quick.json [hunk 1]
```diff
@@ -0,0 +1,22 @@
+{ 
+    "displayNames": { 
+        "_": ""
+    }, 
+    "include": [
+          {
+            "StaticConfigs": {
+              "Ubuntu22": {
+                "OSVmImage": "MMSUbuntu22.04",
```
Comment (weshaggard) [inline eng/pipelines/templates/stages/platform-matrix-quick.json:9]: Do you also want to remove this from the default matrix?

File: eng/pipelines/templates/stages/archetype-sdk-client.yml [hunk 2]
```diff
@@ -88,12 +88,38 @@ stages:
           ArtifactsSource: ${{ parameters.ArtifactsSource }}
           CMakeTestOptions: ${{ parameters.CMakeTestOptions }}
           CMakeSourceTestOptions: ${{ parameters.CMakeSourceTestOptions }}
+  
+  # QuickBuild runs a narrower set of fast builds. If these builds pass, 
+  # Integration can launch immediately without awaiting more build and test jobs
+  - stage: QuickBuild
```
Comment (weshaggard) [inline eng/pipelines/templates/stages/archetype-sdk-client.yml:94]: Is quick build the right name here? That might lead folks to adding more things here. Perhaps PrePublishBuild or something similar to be clearer of the intention.

File: eng/pipelines/templates/stages/platform-matrix-quick.json (hunk 1 above)
Comment (danieljurek) [reply eng/pipelines/templates/stages/platform-matrix-quick.json:9]: It is removed from the other build matrix: https://github.com/Azure/azure-sdk-for-cpp/pull/4522/files#diff-6d57a4119ae22b0fede4d3aab5dff565cee910416d4eaa647d83e1d749c9b559L159

**Comment by danieljurek** [comment]
Successful run -- https://dev.azure.com/azure-sdk/internal/_build/results?buildId=2702916&view=results

**Comment by danieljurek** [comment]
Most recent successful run -- https://dev.azure.com/azure-sdk/internal/_build/results?buildId=2707194&view=results

**Review by benbp — APPROVED** [review]

File: eng/pipelines/templates/stages/platform-matrix-quick.json (hunk 1 above)
Comment (weshaggard) [reply eng/pipelines/templates/stages/platform-matrix-quick.json:9]: Perhaps I'm missing something but that just shows removal of `included: {}` how does that map to this build configuration?

File: eng/pipelines/templates/stages/platform-matrix-quick.json [hunk 3]
```diff
@@ -0,0 +1,22 @@
+{
```
Comment (weshaggard) [inline eng/pipelines/templates/stages/platform-matrix-quick.json:1]: minor but you left this file with the name quick in it.

**Pull request merged**